src/foundry/client/ui/dialog.d.ts

```diff
@@ -58,9 +63,9 @@ declare class Dialog<Options extends DialogOptions = DialogOptions> extends Appl
    */
   static override get defaultOptions(): DialogOptions;
 
-  static get title(): string;
+  override get title(): string;
 
-  override getData(options?: Partial<Options>): MaybePromise<object>;
+  override getData(options?: Partial<Options>): { content: string; buttons: Record<string, Dialog.Button> };
```

Comment:
The reason it was typed that loosely before is polymorphism. `getData` by its nature can return wildly different things down a single inheritance chain. If we would type it concretely, we would sooner or later have typescript yell at us everywhere, that we are violating polymorphism. So at one point we decided to keep this vague for the built-in types, since `getData` is only used to populate a view model. If some system or module author wants to have this more concrete, they can make sure of this in their own overrides and sub classes. So please revert this.